freqtrade/freqtradebot.py

```diff
@@ -1314,17 +1312,17 @@ def execute_trade_exit(
                 logger.exception(f"Could not cancel stoploss order {trade.stoploss_order_id}")
 
         order_type = ordertype or self.strategy.order_types[exit_type]
-        if sell_reason.sell_type == SellType.EMERGENCY_SELL:
+        if exit_reason.exit_type == ExitType.EMERGENCY_EXIT:
             # Emergency sells (default to market!)
-            order_type = self.strategy.order_types.get("emergencysell", "market")
+            order_type = self.strategy.order_types.get("emergencyexit", "market")
 
         amount = self._safe_exit_amount(trade.pair, trade.amount)
         time_in_force = self.strategy.order_time_in_force['sell']  # TODO-lev update to exit
 
         if not strategy_safe_wrapper(self.strategy.confirm_trade_exit, default_retval=True)(
                 pair=trade.pair, trade=trade, order_type=order_type, amount=amount, rate=limit,
-                time_in_force=time_in_force, sell_reason=sell_reason.sell_reason,
-                current_time=datetime.now(timezone.utc)):  # TODO-lev: Update to exit
+                time_in_force=time_in_force, exit_reason=exit_reason.exit_reason,
```

Comment:
We will want to avoid hard breaking changes if possible.

This change breaks every strategy that implements the current interface by no longer providing `sell_reason` - but only once this callback is actually called (so potentially, days into running the new bot version).

One way i can imagine is to provide both `sell_reason` and `exit_reason` for some releases  - marking sell_reason (and whatever else changed) as deprecated in the deprecation docs.